src/services/blocks/BlocksService.ts

```diff
@@ -327,46 +334,68 @@ export class BlocksService extends AbstractService {
 	) {
-		let parentParentHash: Hash;
-		if (block.header.number.toNumber() > 1) {
-			parentParentHash = (await api.rpc.chain.getHeader(parentHash))
-				.parentHash;
+		const perByte = api?.consts?.transactionPayment?.transactionByteFee;
+		const extrinsicBaseWeight = api?.consts?.system?.extrinsicBaseWeight;
+
+		let calcFee, specName, specVersion;
+		if (perByte === undefined || extrinsicBaseWeight === undefined) {
+			// We do not have the neccesary materials to build calcFee, so we just give a dummy function
+			// that aligns with the expected API of calcFee.
+			calcFee = { calc_fee: () => null };
+
+			const version = await api.rpc.state.getRuntimeVersion(parentHash);
+			[specVersion, specName] = [
+				version.specName.toString(),
+				version.specVersion.toNumber(),
+			];
 		} else {
-			parentParentHash = parentHash;
-		}
+			const coefficients = api.consts.transactionPayment.weightToFee.map(
+				(c) => {
+					return {
+						coeffInteger: c.coeffInteger.toString(),
+						coeffFrac: c.coeffFrac,
+						degree: c.degree,
+						negative: c.negative,
+					};
+				}
+			);
 
-		const perByte = api.consts.transactionPayment.transactionByteFee;
-		const extrinsicBaseWeight = api.consts.system.extrinsicBaseWeight;
-		const multiplier = await api.query.transactionPayment.nextFeeMultiplier.at(
-			parentHash
-		);
-		// The block where the runtime is deployed falsely proclaims it would
-		// be already using the new runtime. This workaround therefore uses the
-		// parent of the parent in order to determine the correct runtime under which
-		// this block was produced.
-		const version = await api.rpc.state.getRuntimeVersion(parentParentHash);
-		const specName = version.specName.toString();
-		const specVersion = version.specVersion.toNumber();
-		const coefficients = api.consts.transactionPayment.weightToFee.map(
-			(c) => {
-				return {
-					coeffInteger: c.coeffInteger.toString(),
-					coeffFrac: c.coeffFrac,
-					degree: c.degree,
-					negative: c.negative,
-				};
+			// The block where the runtime is deployed falsely proclaims it would
+			// be already using the new runtime. This workaround therefore uses the
+			// parent of the parent in order to determine the correct runtime under which
+			// this block was produced.
+			let parentParentHash: Hash;
+			if (block.header.number.toNumber() > 1) {
+				parentParentHash = (await api.rpc.chain.getHeader(parentHash))
+					.parentHash;
+			} else {
+				parentParentHash = parentHash;
 			}
-		);
 
-		return {
-			calcFee: CalcFee.from_params(
+			const [version, multiplier] = await Promise.all([
+				api.rpc.state.getRuntimeVersion(parentParentHash),
+				api.query?.transactionPayment?.nextFeeMultiplier?.at(
+					parentHash
+				),
+			]);
+
+			[specName, specVersion] = [
+				version.specName.toString(),
+				version.specVersion.toNumber(),
+			];
+
+			calcFee = CalcFee.from_params(
 				coefficients,
 				BigInt(extrinsicBaseWeight.toString()),
 				multiplier.toString(),
```

Comment:
`multiplier` can be `undefined`, so this can throw on runtime if any of the existence checks at 377 fails, so this can throw on runtime. If the assertion is that we can always get the `multiplier`, we should error at 377 when the error happens rather than kicking the ball down the road.

Reply:
Good catch. Not sure I have run into a scenario where `multiplier` is undefined but `perByte` and `extrinsicBaseWeight` is defined, but good to cover it. I added a check at top to see if the multiplier query is a function